### PR TITLE
feat: activity logging

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
The client's activity pulse logging mechanism. Works in conjunction with #12340

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
This is a prototype, no tests are necessary, at this juncture.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: feat/activity-pulse 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**</details>

### CODE REVIEW

File: app/client/public/logger.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  const timestamp = getCurrentUTCTimestamp(date);
+  return timestamp - (date.getUTCMinutes() * 60 + date.getUTCSeconds());
+}
+
+const PULSE_API_ENDPOINT = "/api/v1/usage-pulse";
+
+// Use standard fetch to POST (fire and forget)
+function sendHTTPPulse() {
+  fetch(PULSE_API_ENDPOINT, {
+    credentials: "same-origin",
```
Comment (sharat87) [inline app/client/public/logger.js:19]:
```suggestion
    method: "POST",
    credentials: "same-origin",
```

File: app/client/public/logger.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+function getCurrentUTCTimestamp(date) {
+  return Math.floor((date || new Date()).getTime() / 1000);
+}
+
+// Gets the unix timestamp of the hour
+// For a current time of 19:15, returns the timestamp of 19:00
+function getCurrentUTCHourTimestamp() {
+  const date = new Date();
+  const timestamp = getCurrentUTCTimestamp(date);
+  return timestamp - (date.getUTCMinutes() * 60 + date.getUTCSeconds());
```
Comment (sharat87) [inline app/client/public/logger.js:11]: I gather from the comments above, that the point  here is to reset minutes and seconds to 0 from current time. Would this be more reliable then?

```suggestion
  date.setUTCMinutes(0);
  date.setUTCSeconds(0);
  return getCurrentUTCTimestamp(date);
```

**Review by sharat87 — CHANGES_REQUESTED** [review]

File: app/client/public/logger.js (hunk 1 above)
Comment (riodeuno) [reply app/client/public/logger.js:19]: Whoops! Good catch! I'll make the change.

File: app/client/public/logger.js (hunk 2 above)
Comment (riodeuno) [reply app/client/public/logger.js:11]: Yes, we can, they both result in the same value.

Screenshot 2022-04-01 at 12 11 45 PM

Comment (sharat87) [reply app/client/public/logger.js]: Got it. No worries then. Less of an issue than the other comment. :)

**Review by sharat87 — APPROVED** [review]

**Pull request merged**